adapter/googleServiceControl/googleServiceControl.go

```diff
@@ -0,0 +1,116 @@
+package googleServiceControl
+
+import (
+	"bytes"
+	"context"
+	"github.com/pborman/uuid"
+	sc "google.golang.org/api/servicecontrol/v1"
+	"istio.io/mixer/adapter/googleServiceControl/config"
+	"istio.io/mixer/pkg/adapter"
+	"istio.io/mixer/template/metric"
+	"time"
+)
+
+type handlerBuilder struct {
+	createClientFn
+}
+
+type serviceControlHandler struct {
+	serviceControlClient *sc.Service
+	env                  adapter.Env
+	configParams         *config.Params
+}
+
+func NewHandlerBuilder() adapter.HandlerBuilder {
+	builder := &handlerBuilder{
+		createClientFn: createClient,
+	}
+	return builder
+}
+
+func (builder *handlerBuilder) Build(c adapter.Config, env adapter.Env) (adapter.Handler, error) {
+	client, err := builder.createClientFn(env.Logger())
+	if err == nil {
+		return nil, err
+	}
+	params := c.(*config.Params)
+	handler := &serviceControlHandler{
+		serviceControlClient: client,
+		env:                  env,
+		configParams:         params,
+	}
+	return handler, nil
+}
+
+func (builder *handlerBuilder) ConfigureMetricHandler(instanceTypes map[string]*metric.Type) error {
+	return nil
+}
+
+func (handler *serviceControlHandler) HandleMetric(ctx context.Context, instances []*metric.Instance) error {
+	reportRequest, err := handleMetric()
+	if err != nil {
+		return err
+	}
+	_, err = handler.serviceControlClient.Services.Report(handler.configParams.ServiceName, reportRequest).Do()
+	return err
+}
+
+func handleMetric() (*sc.ReportRequest, error) {
+	byteBuf := bytes.NewBufferString("mixer-metric-report-id-")
```

Comment:
please follow go convention for naming local variables: https://github.com/golang/go/wiki/CodeReviewComments#variable-names
Suggest renaming such as:
byteBuf -> b
timeNow -> t
operationId -> op

Reply:
receiver and local variables name are shorten but not to sacrifice readability too much.

Reply:
It's on your. But the basic idea is to use package name as a hint for variable name. to avoid redundancies like 
people := &People{...}.
Go also uses tab for indenting. Long names will likely push code to the right on other's tab space settings.